substrate/client/network/benches/request_response_protocol.rs

```diff
@@ -194,29 +230,12 @@ where
 // The libp2p request-response implementation does not provide any backpressure feedback.
 // So this benchmark is useless until we implement it for litep2p.
 #[allow(dead_code)]
-async fn run_with_backpressure<B, H, N>(size: usize, limit: usize)
-where
-	B: BlockT<Hash = H256> + 'static,
-	H: ExHashT,
-	N: NetworkBackend<B, H>,
-{
-	let listen_address1 = get_listen_address();
-	let listen_address2 = get_listen_address();
-	let (worker1, _rx1, _notification_service1) = create_network_worker::<B, H, N>(listen_address1);
-	let service1 = worker1.network_service().clone();
-	let (worker2, rx2, _notification_service2) =
-		create_network_worker::<B, H, N>(listen_address2.clone());
-	let peer_id2 = worker2.network_service().local_peer_id();
-
-	worker1.network_service().add_known_address(peer_id2, listen_address2.into());
-
-	let network1_run = worker1.run();
-	let network2_run = worker2.run();
-	let (break_tx, break_rx) = async_channel::bounded(10);
+async fn run_with_backpressure(setup: Arc<BenchSetup>, size: usize, limit: usize) {
+	let (break_tx, break_rx) = async_channel::bounded(1);
 	let requests = futures::future::join_all((0..limit).into_iter().map(|_| {
```

Comment:
It would be interesting to compare the results when using a single-threaded runtime and a multi-threaded one and see if they differ.